addons/mail/models/mail_channel_partner.py

```diff
@@ -109,3 +88,47 @@ def _get_as_sudo_from_request(self, request, channel_id):
         if guest:
             return guest.env['mail.channel.partner'].sudo().search([('channel_id', '=', channel_id), ('guest_id', '=', guest.id)], limit=1)
         return self.env['mail.channel.partner']
+
+    # --------------------------------------------------------------------------
+    # RTC (voice/video)
+    # --------------------------------------------------------------------------
+
+    def _rtc_join_call(self):
+        self.ensure_one()
+        self._rtc_remove_invitation()
+        self.rtc_session_ids._disconnect()
+        rtc_session = self.env['mail.channel.rtc.session'].create({'channel_partner_id': self.id})
+        self.channel_id._rtc_notify_sessions_change()
+        res = {
+            'iceServers': self.env['mail.ice.server']._get_ice_servers() or False,
+            'rtcSessions': self.channel_id.rtc_session_ids._mail_rtc_session_format_by_channel().get(self.channel_id),
+            'sessionId': rtc_session.id,
+        }
+        if len(self.channel_id.rtc_session_ids) == 1 and self.channel_id.channel_type in {'chat', 'group'}:
+            invited_partners, invited_guests = self.channel_id._invite_members_to_rtc()
+            res.update({
+                'invitedGuests': [('insert', invited_guests)],
+                'invitedPartners': [('insert', invited_partners)],
+            })
+        return res
+
+    def _rtc_leave_call(self):
+        self.ensure_one()
+        self._rtc_remove_invitation()
+        self.rtc_session_ids.unlink()
+        self.channel_id._rtc_notify_sessions_change()
```

Comment:
I think we should not call `_rtc_notify_sessions_change` if there was no `rtc_session_ids` (as long as we use `_rtc_leave_call` to reject invitations).